src/char/int_storage.c

```diff
@@ -240,13 +248,100 @@ int mapif_save_guild_storage_ack(int fd, int account_id, int guild_id, int fail)
 	return 0;
 }
 
+//=========================================================
+// Account Storage
 //---------------------------------------------------------
-// packet from map server
+/**
+ * Parses account storage load request from map server.
+ * @packet 0x3010 [in] <account_id>.L
+ * @param  fd     [in] file/socket descriptor
+ * @return 1 on success, 0 on failure.
+ */
+int mapif_parse_AccountStorageLoad(int fd)
+{
+	int account_id = RFIFOL(fd, 2);
 
+	Assert_ret(fd > 0);
+	Assert_ret(account_id > 0);
+
+	mapif->account_storage_load(fd, account_id);
+
+	return 1;
+}
+
+/**
+ * Loads the account storage and send to the map server.
+ * @packet 0x3805     [out] <account_id>.L <struct item[]>.P
+ * @param  fd         [in]  file/socket descriptor.
+ * @param  account_id [in]  account id of the session.
+ * @return 1 on success, 0 on failure.
+ */
+int mapif_account_storage_load(int fd, int account_id)
+{
+	struct storage_data p = { 0 };
+	int count = 0, i = 0, len = 0;
+
+	Assert_ret(account_id > 0);
+
+	if ((count = inter_storage->fromsql(account_id, &p)) == 0)
+		return 0;
```

Comment:
I don't think an empty storage should be considered a failed load. It should still send the info ("the storage is empty") to the mapserver